### PR TITLE
OPAL-2865 - show message when no VCF store service is configured for …

### PR DESCRIPTION
**Pull request by meek0**
…the project. Add VCF store servicechooser in project's admin

### CODE REVIEW

File: opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectResource.java [hunk 1]
```diff
@@ -133,14 +133,7 @@ public KeyStoreResource getKeyStoreResource() {
...
     Project project = getProject();
-    if (!opalRuntime.hasServicePlugins()) throw new NoSuchServiceException(VCFStoreService.SERVICE_TYPE);
-    if (!project.hasVCFStoreService()) {
-      // for now get the first one. Some day, the service type will be a project admin choice
-      ServicePlugin service = opalRuntime.getServicePlugins()
-          .stream().filter(s -> opalRuntime.isVCFStorePluginService(s)).iterator().next();
-      project.setVCFStoreService(service.getName());
-      projectService.save(project);
-    }
+    if (!opalRuntime.hasServicePlugins() || !project.hasVCFStoreService()) throw new NoSuchServiceException(VCFStoreService.SERVICE_TYPE);
```
Comment (ymarcon) [inline opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectResource.java]: This are different situations:
* there is no service => do not show the genotypes tab
* the project has no store service

**Comment by meek0** [comment]
Hide the "Genotypes" tab when there are no plugins of type vcf-store. In the admin will not show the chooser in the modal for the same reason. But the vcf store service panel in the admin view will still be present.

File: opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectResource.java [hunk 2]
```diff
@@ -134,13 +134,6 @@ public KeyStoreResource getKeyStoreResource() {
   public VCFStoreResource getVCFStoreResource() {
     Project project = getProject();
     if (!opalRuntime.hasServicePlugins()) throw new NoSuchServiceException(VCFStoreService.SERVICE_TYPE);
-    if (!project.hasVCFStoreService()) {
-      // for now get the first one. Some day, the service type will be a project admin choice
-      ServicePlugin service = opalRuntime.getServicePlugins()
-          .stream().filter(s -> opalRuntime.isVCFStorePluginService(s)).iterator().next();
-      project.setVCFStoreService(service.getName());
-      projectService.save(project);
-    }
```
Comment (ymarcon) [inline opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectResource.java]: Why not setting it automatically: the choice you propose is not a real one, as there is only one option!

Comment (meek0) [inline opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectResource.java]: For now, yes, but couldn't there be more than one vcf store plugin in the future?

Comment (meek0) [inline opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectResource.java]: If the genotypes tab is hidden there would also be no way to set it other than doing a query on a project's vcf-store resource.

Comment (ymarcon) [inline opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectResource.java]: * you can check the number of vcf store services if you wish and if it's one... set it.
* protect orientdb with "synchronized"

Comment (meek0) [inline opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectResource.java]: It's not really a problem with synchronised but rather the transaction.

Comment (meek0) [inline opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectResource.java]: Change made to add vcf store service name automatically when a call to the project's vcf-store is made.

**Pull request merged**